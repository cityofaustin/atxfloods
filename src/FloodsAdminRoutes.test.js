import React from 'react';
import ReactDOM from 'react-dom';
import { Route, MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import FloodsAdminRoutes from './FloodsAdminRoutes';
import Public from './Public';
import auth from './services/awsAuth';

jest.mock('./services/awsAuth', () => {
  return {
    isAuthenticated: jest.fn()
  }
});

it('should render the root page correctly when the user is logged out', () => {
  auth.isAuthenticated.mockReturnValue(false);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the root page correctly when the user is logged in', () => {
  auth.isAuthenticated.mockReturnValue(true);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the public page correctly when the user is logged out', () => {
  auth.isAuthenticated.mockReturnValue(false);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/public' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the public page correctly when the user is logged in', () => {
  auth.isAuthenticated.mockReturnValue(true);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/public' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the protected page correctly when the user is logged out', () => {
  auth.isAuthenticated.mockReturnValue(false);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/protected' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the protected page correctly when the user is logged in', () => {
  auth.isAuthenticated.mockReturnValue(true);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/protected' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the create user page correctly when the user is logged out', () => {
  auth.isAuthenticated.mockReturnValue(false);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/createuser' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the create user page correctly when the user is logged in', () => {
  auth.isAuthenticated.mockReturnValue(true);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/createuser' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
