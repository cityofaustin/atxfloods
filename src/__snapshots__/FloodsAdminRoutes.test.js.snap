exports[`test should render the create user page correctly when the user is logged in 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <div>
    <p>
      Create a User
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`test should render the create user page correctly when the user is logged out 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <div>
    <p>
      You must log in to view the page at 
      /createuser
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`test should render the protected page correctly when the user is logged in 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <h3>
    Protected
  </h3>
</div>
`;

exports[`test should render the protected page correctly when the user is logged out 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <div>
    <p>
      You must log in to view the page at 
      /protected
    </p>
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value="" />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="" />
      <input
        type="submit" />
    </form>
  </div>
</div>
`;

exports[`test should render the public page correctly when the user is logged in 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <h3>
    Public
  </h3>
</div>
`;

exports[`test should render the public page correctly when the user is logged out 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
  <h3>
    Public
  </h3>
</div>
`;

exports[`test should render the root page correctly when the user is logged in 1`] = `
<div>
  <p>
    Welcome! 
    <button
      onClick={[Function]}>
      Sign out
    </button>
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
</div>
`;

exports[`test should render the root page correctly when the user is logged out 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
</div>
`;
